src/pages/ShoppingList.js

```diff
@@ -84,26 +101,50 @@ const ShoppingList = ({ token }) => {
             getShoppingList();
         });
     };
+
+    const listFilterChange = e => {
+        let normalizeString = e.target.value
+            .toLowerCase()
+            .replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '')
+            .trim()
+            .replace(/\s{2,}/g, ' ');
+        setFilterString(normalizeString);
+    };
+
+    const filteredList = shoppingListItems.filter(item => {
+        return item.itemName.toLowerCase().includes(filterString.toLowerCase());
+    });
+
     return (
         <div>
+            <label>Search for an item</label>
+            <input
+                type="text"
+                placeholder="Search..."
+                onChange={e => setFilterString(e.target.value)}
+            />
+            <button>X</button>
             <ul>
-				{shoppingListItems.length>0?
-					(shoppingListItems.map(item => (
-						<div>
-							<input
-								key={item.id}
-								id={item.id}
-								type="checkbox"
-								name={item.id}
-								value={item.isChecked}
-								checked={item.isChecked}
-								onChange={handleCheck}
-							/>{' '}
-							{item.itemName}
-						</div>
-                ))):(
-					welcomeInstructions()
-				)}
+                {filterString
+                    ? filteredList.map(item => {
+                          return (
```

Comment:
You could use the same approach here as you are using below down below
`filteredList.map(item => shoppingListItemInput(item))`